src/module.cc

```diff
@@ -0,0 +1,338 @@
+  } else {
+    exit(0); // no core dump requested
+  }
+}
+
+bool OnUncaughtException(v8::Isolate* isolate) {
+   // Trigger NodeReport if required
+  if (nodereport_events & NR_EXCEPTION) {
+    TriggerNodeReport(Isolate::GetCurrent(), kException, "exception", "OnUncaughtException (nodereport/src/module.cc)", NULL);
+  } 
+  if (nodereport_core) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+#ifndef _WIN32
+static void SignalDumpInterruptCallback(Isolate *isolate, void *data) {
+  if (report_signal != 0) {
+    fprintf(stderr,"SignalDumpInterruptCallback - handling signal\n");
+    if (nodereport_events & NR_SIGNAL) {
+      if (nodereport_verbose) {
+        fprintf(stderr,"SignalDumpInterruptCallback - triggering NodeReport\n");
+      }
+      TriggerNodeReport(Isolate::GetCurrent(), kSignal_JS,
+                        node::signo_string(*(static_cast<int *>(data))),
+                        "node::SignalDumpInterruptCallback()", NULL);
+    }
+    report_signal = 0;
+  }
+}
+static void SignalDumpAsyncCallback(uv_async_t* handle) {
+  if (report_signal != 0) {
+    fprintf(stderr,"SignalDumpAsyncCallback - handling signal\n");
+    if (nodereport_events & NR_SIGNAL) {
+      if (nodereport_verbose) {
+        fprintf(stderr,"SignalDumpAsyncCallback - triggering NodeReport\n");
+      }
+      size_t signo_data = reinterpret_cast<size_t>(handle->data);
+      TriggerNodeReport(Isolate::GetCurrent(), kSignal_UV,
+                        node::signo_string(static_cast<int>(signo_data)),
+                        "node::SignalDumpAsyncCallback()", NULL);
+    }
+    report_signal = 0;
+  }
+}
+
+static void RegisterSignalHandler(int signal, void (*handler)(int signal), bool reset_handler = false) {
+  struct sigaction sa;
+  memset(&sa, 0, sizeof(sa));
+  sa.sa_handler = handler;
+  sa.sa_flags = reset_handler ? SA_RESETHAND : 0;
+  sigfillset(&sa.sa_mask);
+  sigaction(signal, &sa, nullptr);
+}
+
+// Raw signal handler for triggering a NodeReport - runs on an arbitrary thread
+static void SignalDump(int signo) {
+  // Check atomic for NodeReport already pending, storing the signal number
+  if (__sync_val_compare_and_swap(&report_signal, 0, signo) == 0) {
+    uv_sem_post(&report_semaphore);  // Hand-off to watchdog thread
+  }
+}
+
+// Watchdog thread implementation for signal-triggered NodeReport
+inline void* ReportSignalThreadMain(void* unused) {
+  for (;;) {
+    uv_sem_wait(&report_semaphore);
+    if (nodereport_verbose) {
+      fprintf(stderr, "Signal %s received by nodereport module\n", node::signo_string(report_signal));
+    }
+    uv_mutex_lock(&node_isolate_mutex);
+    if (auto isolate = node_isolate) {
+      // Request interrupt callback for running JavaScript code
+      isolate->RequestInterrupt(SignalDumpInterruptCallback, &report_signal);
+      // Event loop may be idle, so also request an async callback
+      size_t signo_data = static_cast<size_t>(report_signal);

```

Comment:
I think passing (the address of) `report_signal` as a callback argument just obscures the fact that it's `report_signal`.  I'd drop it.